pkg/scheduler/controller.go

```diff
@@ -0,0 +1,249 @@
+		indexers.ByLogicalClusterPath: indexers.IndexByLogicalClusterPath,
+	})
+
+	edgePlacementAccess.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.enqueueEdgePlacement,
+		UpdateFunc: func(_, newObj interface{}) { c.enqueueEdgePlacement(newObj) },
+		DeleteFunc: c.enqueueEdgePlacement,
+	})
+
+	locationAccess.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: c.enqueueLocation,
+		UpdateFunc: func(old, obj interface{}) {
+			oldLoc := old.(*schedulingv1alpha1.Location)
+			newLoc := obj.(*schedulingv1alpha1.Location)
+			if !apiequality.Semantic.DeepEqual(oldLoc.Spec, newLoc.Spec) || !apiequality.Semantic.DeepEqual(oldLoc.Labels, newLoc.Labels) {
+				c.enqueueLocation(obj)
+			}
+		},
+		DeleteFunc: c.enqueueLocation,
+	})
+
+	syncTargetAccess.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: c.enqueueSyncTarget,
+		UpdateFunc: func(old, obj interface{}) {
+			oldST := old.(*workloadv1alpha1.SyncTarget)
+			newST := obj.(*workloadv1alpha1.SyncTarget)
+			if !apiequality.Semantic.DeepEqual(oldST.Spec, newST.Spec) || !apiequality.Semantic.DeepEqual(oldST.Labels, newST.Labels) {
+				c.enqueueSyncTarget((obj))
+			}
+		},
+		DeleteFunc: c.enqueueSyncTarget,
+	})
+
+	return c, nil
+}
+
+func (c *controller) enqueueEdgePlacement(obj interface{}) {
+	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	klog.FromContext(c.context).V(2).Info("queueing EdgePlacement", "key", key)
+	c.queue.Add(
+		queueItem{
+			triggeringKind: triggeringKindEdgePlacement,
+			key:            key,
+		},
+	)
+}
+
+func (c *controller) enqueueLocation(obj interface{}) {
+	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	klog.FromContext(c.context).V(2).Info("queueing Location", "key", key)
+	c.queue.Add(
+		queueItem{
+			triggeringKind: triggeringKindLocation,
+			key:            key,
+		},
+	)
+}
+
+func (c *controller) enqueueSyncTarget(obj interface{}) {
+	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	klog.FromContext(c.context).V(2).Info("queueing SyncTarget", "key", key)
+	c.queue.Add(
+		queueItem{
+			triggeringKind: triggeringKindSyncTarget,
+			key:            key,
+		},
+	)
+}
+
+// Run starts the controller, which stops when c.context.Done() is closed.
+func (c *controller) Run(numThreads int) {
+	defer runtime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	logger := klog.FromContext(c.context)
+	logger.Info("starting controller")
+	defer logger.Info("shutting down controller")
+
+	for i := 0; i < numThreads; i++ {
+		go wait.UntilWithContext(c.context, c.startWorker, time.Second)
+	}
+
+	<-c.context.Done()
+}
+
+func (c *controller) startWorker(ctx context.Context) {
```

Comment:
This does not "start" a worker, it "runs" a worker.

Reply:
Fixed in https://github.com/kcp-dev/edge-mc/pull/206/commits/59fdfc24abc59e788a7523de8ea557f9839dd4cd.